doodle.py

```diff
@@ -60,26 +61,45 @@ def main():
     create_doodle(options, args)
 
 
+# Parse date string and return date and whether it was relative or not
 def derelativize_date(datestr: str, base=datetime.now()):
     if datestr == '':
-        return base.replace(hour=0, minute=0, second=0, microsecond=0)
+        return base.replace(hour=0, minute=0, second=0, microsecond=0), True
 
     if not datestr.startswith('+'):
         return datetime.strptime(datestr, '%Y-%m-%d')
 
-    return base.replace(hour=0, minute=0, second=0, microsecond=0) + timedelta(days=int(datestr[1:]))
+    return base.replace(hour=0, minute=0, second=0, microsecond=0) + timedelta(days=int(datestr[1:])), True
 
 
 def dates_from_arg(args):
     parts = str(args.dates).split(':')
     if len(parts) != 2:
         raise Exception('Malformed date string. See -h')
 
-    start = derelativize_date(parts[0])
+    start, _ = derelativize_date(parts[0])
+    end, relative = derelativize_date(parts[1], start)
+    ndays = (end - start).days  # Number of days to generate slots on
 
     dates = []
-    for d in range(0, (derelativize_date(parts[1], start) - start).days):
-        dates.append(start + timedelta(days=d))
+
+    i = 0
+    while i < ndays:
```

Comment:
I still think that is better to use for loop
```suggestion
    for i in range(ndays):
```